tools/globcheck/globcheck.py

```diff
@@ -0,0 +1,147 @@
+import os
+import sys
+import argparse
+import warnings
+import pandas as pd
+import logging as log
+import MDAnalysis as mda
+from Bio.PDB import PDBParser
+from cath_alphaflow.models.domains import GeneralDomainID, ChoppingPdbResLabel
+from cath_alphaflow.commands.measure_globularity import (
+    calculate_normed_radius_of_gyration,calculate_packing_density)
+
+# Basic logging configuration
+log.basicConfig(level=log.INFO, \
+                format='%(levelname)s - %(message)s')
+
+# Thresholds
+asphericity_threshold = 0.1
+rg_threshold = 0.356
+pd_threshold = 10.333
+
+def validate_residue_range(pdb_file, trimming_range):
+    """Validates that the provided residue trimming range is 
+    within the actual residue range present in the PDB structure.
+
+    Args:
+        pdb_file (str): Path to the input PDB
+        trimming_range (str): Residue range in format "start-end" ("1-122")
+    """
+    if not os.path.isfile(pdb_file):
+        log.error(f"❌ PDB file '{pdb_file}' does not exist.")
+        sys.exit(1)
+
+    start_res, end_res = map(int, trimming_range.split('-'))
+    u = mda.Universe(pdb_file)
+    residue_ids = [res.resid for res in u.residues]
+    min_res, max_res = min(residue_ids), max(residue_ids)
+
+    if not (min_res <= start_res <= max_res) or not (min_res <= end_res <= max_res):
+        log.error(f"❌ Error: Trimming range {trimming_range} is outside residue range of the PDB {min_res}-{max_res}")
+        sys.exit(1)
+    log.info(f"✅ Residue range {trimming_range} is valid.")
+
+def calculate_asphericity(pdb_file):
+    """Calculates the asphericity of the structure using MDAnalysis.
+
+    Asphericity is a measure of how non-spherical a structure is,
+    with 0 being a perfect sphere.
+
+    Args:
+        pdb_file (str):Path to the input PDB
+
+    Returns:
+        float: Asphericity value for the structure
+    """
+    u = mda.Universe(pdb_file)
+    asphericity = u.atoms.asphericity(pbc=False)
+    log.info(f"✅ Asphericity = {asphericity:.3f}")
+
+    return asphericity
+
+def calculate_globularity_metrics(pdb_file, trimming_range):
+    """Calculates globularity metrics for a trimmed domain using CATH AlphaFlow.
+
+    The metrics include:
+    - Normalized radius of gyration (Rg)
+    - Packing density (PD)
+
+    Args:
+        pdb_file (str): Path to the input PDB
+        trimming_range (str): Residue range in format "start-end" ("1-122")
+
+    Returns:
+        tuple: A tuple (normed_rg, packing_density), both floats
+    """
+    chopping = ChoppingPdbResLabel.from_str(f"{trimming_range}")
```

Comment:
I am a bit confused by what this does - do we use `calculate_asphericity` on the full structure, but then we trim it additionally (if the PDB is > than the trimming interval) to calculate globularity?